### PR TITLE
bind wlr_xdg_activation_v1.h

### PR DESCRIPTION
**Pull request by novakne**

### CODE REVIEW

File: src/types/xdg_activation_v1.zig [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    extern fn wlr_xdg_activation_v1_create(server: *wl.Server) ?*XdgActivationV1;
+    pub const create = wlr_xdg_activation_v1_create;
+};
+
+pub const XdgActivationTokenV1 = extern struct {
+    activation: *wlr.XdgActivationV1,
+    surface: ?*wlr.Surface,
+    seat: ?*wlr.Seat,
+
+    serial: ?u32, // Invalid if seat is null
```
Comment (ifreund) [inline src/types/xdg_activation_v1.zig:39]: A `?u32` does not have the same ABI as a `u32` and I'm pretty sure won't even compile in an `extern struct` as it doesn't have a well defined memory layout.
```suggestion
    /// Invalid if seat is null
    serial: u32,
```

File: src/types/xdg_activation_v1.zig [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    app_id: ?[*:0]u8,
+
+    link: wl.list.Link,
+
+    // Private state
+    token: [*:0]u8,
+    resource: ?*wl.Resource,
+    timeout: ?*wl.EventSource, // TODO: Not sure about this
```
Comment (ifreund) [inline src/types/xdg_activation_v1.zig:47]: This is fine
```suggestion
    timeout: ?*wl.EventSource,
```

File: src/types/xdg_activation_v1.zig [hunk 3]
```diff
@@ -0,0 +1,51 @@
+const wlr = @import("../wlroots.zig");
+
+const wayland = @import("wayland");
+const wl = wayland.server.wl;
+
+pub const XdgActivationV1 = extern struct {
+    // token timeout in milliseconds (0 to disable)
```
Comment (ifreund) [inline src/types/xdg_activation_v1.zig:7]:
```suggestion
    /// token timeout in milliseconds (0 to disable)
```

File: src/types/xdg_activation_v1.zig [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        destroy: wl.Signal(*XdgActivationV1),
+        request_activate: wl.Signal(*XdgActivationV1.RequestActivate.event),
+    },
+
+    // Private state
+    global: *wl.Global,
+
+    server_destroy: wl.Listener(*wl.Server),
+
+    pub const RequestActivate = extern struct {
```
Comment (ifreund) [inline src/types/xdg_activation_v1.zig:22]: The pattern used elsewhere in zig-wlroots is to place this before the first field of `XdgActivationV1` and to do:
```zig
pub const event = struct {
    pub const RequestActivate = extern struct { ... };
};
```

File: src/types/xdg_activation_v1.zig [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        event: extern struct {
+            activation: *wlr.XdgActivationV1,
+            token: *wlr.XdgActivationTokenV1,
+            surface: *wlr.Surface,
+        },
+    };
+
+    extern fn wlr_xdg_activation_v1_create(server: *wl.Server) ?*XdgActivationV1;
+    pub const create = wlr_xdg_activation_v1_create;
```
Comment (ifreund) [inline src/types/xdg_activation_v1.zig:31]: A wrapper here returning `error.OutOfMemory` if the wlroots function returns null wouldn't hurt.

**Review by ifreund — CHANGES_REQUESTED** [review]

File: src/types/xdg_activation_v1.zig (hunk 5 above)
Comment (novakne) [reply src/types/xdg_activation_v1.zig:31]: oops my bad, no idea why I forgot this

**Review by ifreund — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**